specifyweb/frontend/js_src/lib/components/Statistics/index.tsx

```diff
@@ -0,0 +1,599 @@
+          label: queryData.name,
+        })),
+      [queryId]
+    ),
+    false
+  );
+  return data;
+}
+
+function DefaultStatItem({
+  statsSpec,
+  pageName,
+  categoryName,
+  itemName,
+  onRemove: handleRemove,
+  onClick: handleClick,
+}: {
+  readonly statsSpec: IR<
+    IR<{
+      readonly label: string;
+      readonly items: StatCategoryReturn;
+    }>
+  >;
+  readonly pageName: string;
+  readonly categoryName: keyof typeof statsSpec;
+  readonly itemName: string;
+  readonly onRemove: (() => void) | undefined;
+  readonly onClick: (() => void) | undefined;
+}): JSX.Element {
+  const statSpecItemPage = statsSpec[pageName];
+  const statSpecItemObject = statSpecItemPage[categoryName]?.items;
+  const statSpecItem =
+    statSpecItemObject === undefined ? undefined : statSpecItemObject[itemName];
+  const statValue =
+    statSpecItem === undefined ? undefined : statSpecItem.spec.type ===
+      'Querybuildstat' ? (
+      <QueryBuilderStat
+        fields={statSpecItem.spec.fields}
+        statLabel={statSpecItem.label}
+        tableName={statSpecItem.spec.tableName}
+        onClick={handleClick}
+        onRemove={handleRemove}
+      />
+    ) : (
+      <StatsResult
+        query={undefined}
+        statLabel={statSpecItem?.label}
+        statValue={statSpecItem.spec.value}
+        onClick={handleClick}
+        onRemove={handleRemove}
+      />
+    );
+  return statValue ?? <p> {commonText('loading')}</p>;
+}
+
+function CustomStatItem({
+  queryId,
+  onRemove: handleRemove,
+  onClick: handleClick,
+}: {
+  readonly queryId: number;
+  readonly onRemove: (() => void) | undefined;
+  readonly onClick: (() => void) | undefined;
+}): JSX.Element {
+  const { tableName, fields, label } =
+    useCustomStatQueryBuilderSpec(queryId) ?? {};
+  const statValue =
+    tableName === undefined ||
+    fields === undefined ||
+    label === undefined ? undefined : (
+      <QueryBuilderStat
+        fields={fields}
+        statLabel={label}
+        tableName={tableName}
+        onClick={handleClick}
+        onRemove={handleRemove}
+      />
+    );
+  return statValue ?? <p>{commonText('loading')}</p>;
+}
+
+function QueryBuilderStat({
+  tableName,
+  fields,
+  statLabel,
+  onRemove: handleRemove,
+  onClick: handleClick,
+}: {
+  readonly tableName: keyof Tables;
+  readonly fields: RA<
+    Partial<SerializedResource<SpQueryField>> & { readonly path: string }
+  >;
+  readonly statLabel: string;
+  readonly onRemove: (() => void) | undefined;
+  readonly onClick: (() => void) | undefined;
+}): JSX.Element {
+  const frontEndQuery = useFrontEndStatsQuery(tableName, fields);
+  const frontEndStatValue = useFrontEndStat(frontEndQuery);
+  return frontEndStatValue === undefined ? (
+    <p>{commonText('loading')}</p>
+  ) : (
+    <StatsResult
+      query={frontEndQuery}
+      statLabel={statLabel}
+      statValue={frontEndStatValue}
+      onClick={handleClick}
+      onRemove={handleRemove}
+    />
+  );
+}
```

Comment:
Split this file into several smaller ones (figure out which sections you can split it into)

<details>
  <summary>Example solution</summary>
  
  Put all hooks into `hooks.tsx`. Leave the root component in this file. Move all functions that render individual stat items into a separate file. `StatsCategories` and Adding dialog could go into yet one more file. 
  
</details>